packages/dds/merge-tree/src/client.ts

```diff
@@ -346,6 +350,49 @@ export class Client {
         return this.mergeTree.getMarkerFromId(id);
     }
 
+    /**
+     * Revert an op
+     */
+    public rollback?(op: any, localOpMetadata: unknown) {
+        if (op.type !== MergeTreeDeltaType.INSERT) {
+            throw new Error("Unsupported op type for rollback");
+        }
+
+        const lastList = this.mergeTree.pendingSegments?.prev;
+        if (!lastList) {
+            throw new Error("No pending segments");
+        }
+        ListRemoveEntry(lastList);
```

Comment:
Maybe this functionality of popping the last item off the list should be moved to List.